Assets/Global/Scripts/UIScripts/Healthbar.cs

```diff
@@ -4,19 +4,40 @@
 
 public class Healthbar : MonoBehaviour
 {
-    private Image healthbarOverlay;
-    private Image healthbarUnderlay;
+    private Transform HealthbarUnderlayTransform;
+    private Transform HealthbarOverlayTransform;
+    private float originalWidth;
     // Start is called once before the first execution of Update after the MonoBehaviour is created
-    void Start()
+    void Awake()
     {
-        healthbarOverlay = transform.GetChild(1).GetComponent<Image>();
-        healthbarUnderlay = transform.GetChild(0).GetComponent<Image>();
+        HealthbarUnderlayTransform = transform.GetChild(0);
+        HealthbarOverlayTransform = transform.GetChild(1);
+
+        originalWidth = (HealthbarUnderlayTransform as RectTransform).sizeDelta.x;
+
+        UpdateMaxHealth();
     }
 
-    public void UpdateHealthBar(float min, float max, float current)
+    public void UpdateHealthBar(float current)
     {
-        float fillAmount = math.lerp(min, max, current / max);
-        healthbarOverlay.fillAmount = fillAmount / 100;
-        healthbarUnderlay.fillAmount = max / 100;
+        (HealthbarOverlayTransform as RectTransform).sizeDelta = new Vector2(
+            originalWidth * (current / 2), 
+            (HealthbarOverlayTransform as RectTransform).sizeDelta.y
+        );
+    }
+
+    public void UpdateMaxHealth() {
+        Player player = GlobalReference.GetReference<PlayerReference>().Player;
+
+        // doing maxHealth / 2 because 1hp is a half heart
+        (HealthbarUnderlayTransform as RectTransform).sizeDelta = new Vector2(
+            originalWidth * (player.playerStatistic.MaxHealth.GetValue() / 2), 
+            (HealthbarUnderlayTransform as RectTransform).sizeDelta.y
+        );
+
+        (HealthbarOverlayTransform as RectTransform).sizeDelta = new Vector2(
+            originalWidth * (player.playerStatistic.MaxHealth.GetValue() / 2), 
```

Comment:
you are updating the currentHealth with the max health value here.
be carefull with that.